src/sema/namespace.rs

```diff
@@ -309,36 +309,25 @@ impl Namespace {
     pub fn resolve_contract_with_namespace(
         &mut self,
         file_no: usize,
-        expr: &pt::Expression,
+        name: &pt::IdentifierPath,
         diagnostics: &mut Vec<Diagnostic>,
     ) -> Result<usize, ()> {
-        let (namespace, id, dimensions) = self.expr_to_type(file_no, None, expr, diagnostics)?;
-
-        if !dimensions.is_empty() {
+        let (id, namespace) = if let Some((id, namespace)) = name.identifiers.split_last() {
+            (id, namespace.iter().collect())
+        } else {
```

Comment:
The else should never occur. The `name.identifiers` should have at least one element, or there is a bug in the parser.

This might was well be an `.unwrap()` and get rid of the diagnostic.